### ISSUE TITLE
Depthwise Conv1d performance (a naive CUDA kernel is 10x faster)

### ISSUE THREAD

**Issue by BlinkDL** [pytorch/pytorch#75747]
### 🚀 The feature, motivation and pitch

Please improve the CUDA performance of Depthwise Conv1d :)

FYI, I write a naive CUDA kernel and it's already 10x faster than pytorch:

https://github.com/BlinkDL/RWKV-CUDA

RTX3090:
pytorch = fwd 14ms bwd 65ms
CUDA kernel v3 = fwd 0.8ms bwd 5.5ms

**Comment by ipostr08** [comment]
Yes, to be more general, group convolutions are very slow. Nothing has been done to fix it despite many people asking for a fix over the years. E.g. https://github.com/pytorch/pytorch/issues/73764, https://github.com/pytorch/pytorch/issues/18631, https://github.com/pytorch/pytorch/issues/70954, https://discuss.pytorch.org/t/group-convolution-takes-much-longer-than-normal-convolution/92214.

**Comment by songyuc** [comment]
> ### 🚀 The feature, motivation and pitch
> 
> Please improve the CUDA performance of Depthwise Conv1d :)
> 
> 
> 
> FYI, I write a naive CUDA kernel and it's already 10x faster than pytorch:
> 
> 
> 
> https://github.com/BlinkDL/RWKV-CUDA
> 
> 
> 
> RTX3090:
> 
> pytorch = fwd 14ms bwd 65ms
> 
> CUDA kernel v3 = fwd 0.8ms bwd 5.5ms
> 
> ### Alternatives
> 
> _No response_
> 
> ### Additional context
> 
> _No response_
> 
> cc @ngimel

Would you consider make a PR about it?😀

**Comment by JulienSiems** [comment]
@BlinkDL Same here! Would be very interested in a PR and ready to help!

**Comment by BlinkDL** [comment]
> @BlinkDL Same here! Would be very interested in a PR and ready to help!

Hi you can work on my code :)
